src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/GenericTypeParameterBuilderImpl.cs

```diff
@@ -0,0 +1,148 @@
+    {
+        private readonly string _name;
+        private readonly TypeBuilderImpl _type;
+        private readonly int _genParamPosition;
+        private GenericParameterAttributes _genParamAttributes;
+        private bool _isGenericType;
+        [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)]
+        private Type? _parent;
+
+        internal List<CustomAttributeWrapper>? _customAttributes;
+        internal List<Type>? _interfaces;
+
+        internal GenericTypeParameterBuilderImpl(string name, int genParamPosition, TypeBuilderImpl typeBuilder)
+        {
+            _name = name;
+            _genParamPosition = genParamPosition;
+            _type = typeBuilder;
+            _isGenericType = true;
+        }
+
+        protected override void SetBaseTypeConstraintCore([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] Type? baseTypeConstraint)
+        {
+            _parent = baseTypeConstraint;
+
+            if (_parent != null)
+            {
+                _interfaces ??= new List<Type>();
+                _interfaces.Add(_parent);
+            }
+        }
+
+        protected override void SetCustomAttributeCore(ConstructorInfo con, ReadOnlySpan<byte> binaryAttribute)
+        {
+            _customAttributes ??= new List<CustomAttributeWrapper>();
+            _customAttributes.Add(new CustomAttributeWrapper(con, binaryAttribute));
+        }
+
+        protected override void SetGenericParameterAttributesCore(GenericParameterAttributes genericParameterAttributes) =>
+            _genParamAttributes = genericParameterAttributes;
+
+        protected override void SetInterfaceConstraintsCore(params Type[]? interfaceConstraints)
+        {
+            if (interfaceConstraints != null)
+            {
+                _interfaces ??= new List<Type>(interfaceConstraints.Length);
+                _interfaces.AddRange(interfaceConstraints);
+            }
+        }
+        public override Type[] GetGenericParameterConstraints() =>
+            _interfaces == null ? EmptyTypes : _interfaces.ToArray();
+        public override bool IsGenericTypeParameter => _isGenericType;
+        public override bool IsGenericMethodParameter => !_isGenericType;
+        public override int GenericParameterPosition => _genParamPosition;
+        public override GenericParameterAttributes GenericParameterAttributes => _genParamAttributes;
+        public override string Name => _name;
+        public override Module Module => _type.Module;
+        public override Assembly Assembly => _type.Assembly;
+        public override string? FullName => null;
+        public override string? Namespace => null;
+        public override string? AssemblyQualifiedName => null;
+        public override Type UnderlyingSystemType => this;
+        public override bool IsGenericTypeDefinition => false;
+        public override bool IsGenericType => false;
```

Comment:
Shouldn't this be?

```suggestion
        public override bool IsGenericType => _isGenericType;
```

Reply:
Hmmm. I see this used in other places. Perhaps a rename of the member to avoid the ambiguity in this case.

Reply:
> Shouldn't this be?

I guess, in case of nested type parameters

Reply:
> Shouldn't this be?

@AaronRobinsonMSFT when it could be true?

Reply:
> Hmmm. I see this used in other places. Perhaps a rename of the member to avoid the ambiguity in this case.

Renamed